tools/linker/MobileResolveMainAssemblyStep.cs

```diff
@@ -32,8 +35,16 @@ protected override void Process ()
 			Annotations.Push (assembly);
 
 			foreach (var t in assembly.MainModule.Types) {
-				if (!t.HasCustomAttribute (Namespaces.Foundation, "RegisterAttribute"))
+				if (t.IsPublic && embeddinator) {
+					// Mark all public types when in embeddinator mode.
+					// There may be no types with the Register attribute,
+					// which means that without this the assembly might be completely ignored even if it's not linked.
+					Annotations.Mark (t);
```

Comment:
not sure it's (all) needed - but we can revisit that later (no harm in extra annotations)